File: Colage/Features/Auth/Screens/WelcomeScreen.swift
import SwiftUI

struct WelcomeScreen: View {
    let onContinue: () -> Void
    let onLogin: () -> Void

    var body: some View {
        ZStack {
            ColageColors.background.ignoresSafeArea()

            VStack(spacing: 0) {
                Spacer()

                // Logo area
                VStack(spacing: 20) {
                    ZStack {
                        Circle()
                            .fill(ColageColors.primary.opacity(0.15))
                            .frame(width: 120, height: 120)
                        Image(systemName: "location.circle.fill")
                            .font(.system(size: 60))
                            .foregroundStyle(ColageColors.primary)
                    }

                    Text("colage")
                        .font(.system(size: 42, weight: .bold, design: .rounded))
                        .foregroundStyle(ColageColors.textPrimary)

                    Text("discover your campus")
                        .font(ColageFonts.title3)
                        .foregroundStyle(ColageColors.textSecondary)
                }

                Spacer()

                // Features
                VStack(spacing: 16) {
                    FeatureRow(
                        icon: "map.fill",
                        title: "Live Map",
                        subtitle: "See who's around you in real-time"
                    )
                    FeatureRow(
                        icon: "person.2.fill",
                        title: "Student Profiles",
                        subtitle: "Connect via social links"
                    )
                    FeatureRow(
                        icon: "arkit",
                        title: "AR Discovery",
                        subtitle: "Find people through your camera"
                    )
                }
                .padding(.horizontal, 32)

                Spacer()

                // CTAs
                VStack(spacing: 12) {
                    ColagePrimaryButton(title: "Get Started", action: onContinue)

                    Button(action: onLogin) {
                        Text("Already have an account? ")
                            .font(ColageFonts.body)
                            .foregroundStyle(ColageColors.textSecondary)
                        + Text("Log In")
                            .font(ColageFonts.bodyBold)
                            .foregroundStyle(ColageColors.primary)
                    }

                    Text("For verified .edu students only")
                        .font(ColageFonts.caption)
                        .foregroundStyle(ColageColors.textTertiary)
                        .padding(.top, 4)
                }
                .padding(.horizontal, 24)
                .padding(.bottom, 40)
            }
        }
        .navigationBarHidden(true)
    }
}

struct FeatureRow: View {
    let icon: String
    let title: String
    let subtitle: String

    var body: some View {
        HStack(spacing: 16) {
            Image(systemName: icon)
                .font(.system(size: 22))
                .foregroundStyle(ColageColors.primary)
                .frame(width: 44, height: 44)
                .background(ColageColors.primary.opacity(0.12))
                .clipShape(RoundedRectangle(cornerRadius: 12))

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(ColageFonts.bodyBold)
                    .foregroundStyle(ColageColors.textPrimary)
                Text(subtitle)
                    .font(ColageFonts.footnote)
                    .foregroundStyle(ColageColors.textSecondary)
            }

            Spacer()
        }
    }
}

#Preview {
    WelcomeScreen(onContinue: {}, onLogin: {})
}

File: android/app/src/main/java/com/colageclub/colage/features/auth/screens/WelcomeScreen.kt
package com.colageclub.colage.features.auth.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.SpanStyle
import androidx.compose.ui.text.buildAnnotatedString
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.withStyle
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.colageclub.colage.core.design.ColageColors
import com.colageclub.colage.core.design.LocalThemeColor
import com.colageclub.colage.core.design.ColageFonts
import com.colageclub.colage.core.design.ColagePrimaryButton

@Composable
fun WelcomeScreen(
    onGetStarted: () -> Unit,
    onLogin: () -> Unit
) {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .background(ColageColors.Background)
    ) {
        Column(
            modifier = Modifier
                .fillMaxSize()
                .systemBarsPadding(),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(Modifier.weight(1f))

            // Logo area
            Column(
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(20.dp)
            ) {
                Box(
                    modifier = Modifier
                        .size(120.dp)
                        .clip(CircleShape)
                        .background(LocalThemeColor.current.copy(alpha = 0.15f)),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.LocationOn,
                        contentDescription = null,
                        tint = LocalThemeColor.current,
                        modifier = Modifier.size(60.dp)
                    )
                }

                Text(
                    text = "colage",
                    style = ColageFonts.LargeTitle.copy(
                        fontWeight = FontWeight.Bold,
                        fontSize = 42.sp,
                        color = ColageColors.TextPrimary
                    )
                )

                Text(
                    text = "discover your campus",
                    style = ColageFonts.Title3.copy(color = ColageColors.TextSecondary)
                )
            }

            Spacer(Modifier.weight(1f))

            // Feature rows
            Column(
                modifier = Modifier.padding(horizontal = 32.dp),
                verticalArrangement = Arrangement.spacedBy(16.dp)
            ) {
                FeatureRow(
                    icon = Icons.Default.Map,
                    title = "Live Map",
                    subtitle = "See who's around you in real-time"
                )
                FeatureRow(
                    icon = Icons.Default.Group,
                    title = "Student Profiles",
                    subtitle = "Connect via social links"
                )
                FeatureRow(
                    icon = Icons.Default.ViewInAr,
                    title = "AR Discovery",
                    subtitle = "Find people through your camera"
                )
            }

            Spacer(Modifier.weight(1f))

            // CTAs
            Column(
                modifier = Modifier.padding(horizontal = 24.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                ColagePrimaryButton(
                    title = "Get Started",
                    onClick = onGetStarted
                )

                TextButton(onClick = onLogin) {
                    Text(
                        text = buildAnnotatedString {
                            withStyle(SpanStyle(color = ColageColors.TextSecondary)) {
                                append("Already have an account? ")
                            }
                            withStyle(SpanStyle(color = LocalThemeColor.current, fontWeight = FontWeight.SemiBold)) {
                                append("Log In")
                            }
                        },
                        style = ColageFonts.Body
                    )
                }

                Text(
                    text = "For verified .edu students only",
                    style = ColageFonts.Caption.copy(color = ColageColors.TextTertiary)
                )

                Spacer(Modifier.height(16.dp))
            }
        }
    }
}

@Composable
private fun FeatureRow(
    icon: ImageVector,
    title: String,
    subtitle: String
) {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Box(
            modifier = Modifier
                .size(44.dp)
                .clip(RoundedCornerShape(12.dp))
                .background(LocalThemeColor.current.copy(alpha = 0.12f)),
            contentAlignment = Alignment.Center
        ) {
            Icon(
                imageVector = icon,
                contentDescription = null,
                tint = LocalThemeColor.current,
                modifier = Modifier.size(22.dp)
            )
        }

        Column(verticalArrangement = Arrangement.spacedBy(2.dp)) {
            Text(text = title, style = ColageFonts.BodyBold.copy(color = ColageColors.TextPrimary))
            Text(text = subtitle, style = ColageFonts.Footnote.copy(color = ColageColors.TextSecondary))
        }

        Spacer(Modifier.weight(1f))
    }
}
